pkg/scheduler/dependency_graph.go

```diff
@@ -338,29 +368,44 @@ func (rl *ReplicaList) Swap(i, j int) {
 	rl.Items[i], rl.Items[j] = rl.Items[j], rl.Items[i]
 }
 
-func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]string, error) {
-	label := labels.Set{"flow": flow.Name}
+func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]client.Replica, []client.Replica, error) {
+	name := flow.Name
+	if options.FlowInstanceName != "" {
+		name = options.FlowInstanceName
+	}
+	label := labels.Set{"flow": name}
 	var tmp ReplicaList
```

Comment:
It's from other commit, but the name does not reflect for what this is used later.
Why not use there `existingReplicas` as name, and later get from Replicas().List() `tmp` casted after error handling to `existingReplicas`?